ingest/src/main/scala/hydra.ingest/http/IngestionEndpoint.scala

```diff
@@ -80,7 +81,11 @@ class IngestionEndpoint(implicit val system: ActorSystem, implicit val e: Execut
   }
 
   private def exceptionHandler = ExceptionHandler {
-    case e: IllegalArgumentException => complete(400, GenericError(400, e.getMessage))
+    case e: IllegalArgumentException =>
+      if(applicationConfig.get[Boolean]("hydra.ingest.shouldLog400s").valueOrElse(false)) {
+        log.error("Ingestion 400 ERROR: " + e.getMessage)
```

Comment:
Use scala's string interpolation whenever possible. Here that would look like: s"Ingestion 400 ERROR: ${e.getMessage}". It's much more performant than the string concatenation and is easier to read for complicated strings.